File: Ruby/civitas/pkg/civitas-0.0.1/lib/casilla_impuesto.rb

#encoding: utf­8
module Civitas
  class CasillaImpuesto < Casilla
    
    def initialize(_cantidad, _nombre)
      super(_nombre)
      @importe = _cantidad
    end
  
  def recibe_jugador_impuesto(iactual,todos)
     if(jugador_correcto(iactual,todos))
       informe(iactual, todos)
       todos.at(iactual).paga_impuesto(@importe)
     end
  end
    
  def to_sc()
    return "El importe es " + @importe.to_s()
  end
end
  end

File: Java/civitas/src/civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;

public class CasillaImpuesto extends Casilla{
    
    private float importe;
    
    CasillaImpuesto(float _cantidad, String _nombre){
        super(_nombre);
        importe = _cantidad;
    }
    
    @Override
    public void recibeJugador(int iActual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(iActual, todos)){
            super.informe(iActual, todos);
            todos.get(iActual).pagaImpuesto(importe);
        }
    }
    
    @Override
    public String toString(){
        return "\nHas caido en "+ super.getNombre() +" y es un Impuesto de: " + Float.toString(importe);
    }
}
